uips/uip-8.md

```diff
@@ -0,0 +1,209 @@
+
+`nf = hash_3(ds, (nk, cm, pos))`
+
+where the position `pos = 0`. The domain separator `ds` used for nullifier
+derivation is computed as:
+
+```
+ds = from_le_bytes(BLAKE2b-512(b"penumbra.nullifier")) mod q
+```
+
+The resultant nullifier $nf$ is provided as part of the attestation.
+
+##### Randomized Verification Key
+
+The randomized verification key $rk$ is provided as part of the attestation.
+
+#### Witness
+
+##### Merkle Proof of Inclusion
+
+By using a zero value note, the note will be treated as a dummy note in the `SpendCircuit`. This means signers/verifiers do not need to maintain state or provide a valid Merkle authentication path or anchor.
+
+When generating the Merkle proof, clients can use an empty Merkle tree.
+
+##### Note
+
+The witnessed note has zero value, an Rseed consisting of zero bytes [0u8; 32], the fixed asset ID of the Penumbra staking token, and the address that is being attested to.
+
+##### Balance blinding factor
+
+The blinding factor must be zero.
+
+##### Spend authorization randomizer
+
+The randomizer $\alpha$ is used for generating the randomized spend authorization key.
+
+##### Spend verification key
+
+The spend verification key $ak$ corresponds to the spend key that controls the note.
+
+##### Nullifier-deriving key
+
+The nullifier-deriving key $nk$ corresponds to the spend key that controls the note.
+
+### Attestation Format
+
+The attestation $a$ consists of:
+
+* a 64 bytes SpendAuth signature $\sigma$ derived as described above,
+* a 32 byte randomized verification key $rk$,
+* a 32 byte nullifier $nf$,
+* a 192 byte spend ZKP $\pi$.
+
+These four components are concatenated into a 320 byte array and provided to verifiers:
+
+$a = \sigma || rk || nf || \pi$
+
+Verifiers need the verification key for the spend ZKP (provided in the `penumbra-proof-params` crate) and the address being attested to in order to verify this attestation.
+
+### Custody RPCs
+
+The `CustodyService` should be extended with a new RPC method for address attestations:
+
+```protobuf
+service CustodyService {
+  // ...
+  // Create an address attestation for the provided message
+  rpc CreateAddressAttestation(CreateAddressAttestationRequest) returns (CreateAddressAttestationResponse);
+}
+
+message CreateAddressAttestationRequest {
+  // The message to attest to as a google.protobuf.Any
+  google.protobuf.Any message = 1;
+  // The address to create the attestation with
+  Address address = 2;
+}
+
+message CreateAddressAttestationResponse {
+  // The resulting attestation
+  bytes attestation = 1;
+}
+```
+
+The message field MUST be a `google.protobuf.Any` rather than raw bytes. This allows:
+1. Custody backends to decode and understand specific message types they support
+2. Wallets to implement domain-specific user confirmation flows (e.g. "You are attesting that you control this address for purpose X")
```

Comment:
this is a good idea, there should be a description of what the request is actually attesting (is it the byte encoding of the Any?)